### PR TITLE
[EC-355] Move networking resources to modules

### PR DESCRIPTION
**Pull request by christian-calabrese**
### List of changes

Move resources from legacy `core` folder to the new one organizing files in modules, to reuse them across two regions

### Motivation and context

As discussed several times, core TF configuration is too big and must be split through different configs. Moreover, we need to manage two regions with similar resources plus global regions. This file structure makes it easier reusing same definitions. Resources that will be included in this config are listed in Confluence RFC. Everything else will be moved apart

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

File: src/core/_modules/global/modules/dns/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
+output "public_dns_zones" {
+  value = {
+    io = {
+      name = length(azurerm_dns_zone.io_pagopa_it) > 0 ? azurerm_dns_zone.io_pagopa_it[0].name : null
```
Comment (Krusty93) [inline src/core/_modules/global/modules/dns/outputs.tf:4]: Do you think we can remove all the `count`s? I can't see how those conditions could be false

File: src/core/prod/locals.tf [hunk 2]
```diff
@@ -1,21 +1,18 @@
 locals {
   prefix                   = "io"
   env_short                = "p"
-  location_short           = "itn"
+  location_short           = { westeurope = "weu", italynorth = "itn" }
   secondary_location_short = "gwc"
```
Comment (Krusty93) [inline src/core/prod/locals.tf:5]: I'd do the same as you already done with `location_short`, including `gwc` and `neu`

**Review by Krusty93 — APPROVED** [review]
a couple of suggestions

File: src/core/_modules/global/modules/dns/outputs.tf (hunk 1 above)
Comment (christian-calabrese) [reply src/core/_modules/global/modules/dns/outputs.tf:4]: Agree! In fact, I don't know why counts are also present on the resources. What do you think if we remove them also there?
I guess it's been done to allow deployment in other environments, but if it's not a practice anymore, I would remove them

Comment (Krusty93) [reply src/core/_modules/global/modules/dns/outputs.tf:4]: I agree, proceed (even in the case of multiple environments, they would be managed differently now)

**Review by francescopersico — CHANGES_REQUESTED** [review]
Fill the PR template to highlight possible impacts on users.

**Review by francescopersico — APPROVED** [review]

**Pull request merged**